python/mlx/optimizers.py

```diff
@@ -74,6 +82,27 @@ def apply_single(
         update."""
         raise NotImplementedError()
 
+class GD(Optimizer):
+    r"""Gradient descent optimizer.
+    
+    Updates a parameter :math:`w` with a gradient :math:`g` as follows
+    
+    .. math::
+        
+            w_{t+1} = w_t - \lambda g_t
+    Args:
+        learning_rate (float): The learning :math:`\lambda` for the update
+    """
+    def __init__(self, learning_rate: float):
+        super().__init__()
+        
+        self.learning_rate = learning_rate
+    def apply_single(
+        self, gradient: mx.array, parameter: mx.array, state: OptimizerState
+    ):
+        """Performs the GD parameter update
+        """
+        return parameter - self.learning_rate * gradient
```

Comment:
Let's remove this as it's the same as SGD with `momentum=0` which is the default. (If you like you could update the docstring there to make that a little more evident.